src/Containers/HighlightedText/HighlightedText.tsx

```diff
@@ -36,42 +45,62 @@ export const HighlightedText: React.FC<HighlightedTextProps> = ({
     labels,
     ...props
 }): React.ReactElement => {
+    const [numEntries, setNumEntries] = useState(0);
+    const [numNewEntries, setNumNewEntries] = useState(0);
+
+    if (labels.length !== numEntries) {
+        setNumNewEntries (labels.length - numEntries)
+        setNumEntries(labels.length)
+    }
```

Comment:
Why is this not in a function? It will execute every re render . Maybe put it in a useEffect?